### ISSUE TITLE
Accept stdin as a "data values file"

### ISSUE THREAD

**Issue by pivotaljohn** [k14s/ytt#486]
**Describe the problem/challenge you have**
I want to supply a plain YAML data values file on "standard input" (aka stdin) to my `ytt` invocation

(ex: https://kubernetes.slack.com/archives/CH8KCCKA5/p1631819726297200)

**Describe the solution you'd like**

```bash
$ generate-data-values | ytt -f config/ --data-values-file=-
```
where:
- `generate-data-values` is some process that yields (to stdout) a plain YAML that is suitable as a data values file.
- 🛑 If stdin is supplied both for a `--file` entry _and_ `--data-values-file`, this should error out explaining that standard input can only be used once.

**Anything else you would like to add:**
- 🏅 This (subtly) also improves the use of `ytt` in Windows environment where the same line can be used in Powershell.
- 🤔 Consider [the suggestion](https://kubernetes.slack.com/archives/CH8KCCKA5/p1631899266320900?thread_ts=1631819726.297200&cid=CH8KCCKA5) to explore how to reduce the total code paths while implementing this feature: can this kind of file handling (regardless of which flag is accepting the file) be centralized/turned into a core feature of the system?

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by pivotaljohn** [comment]
Maintainers have committed to staff this work. Changed priority accordingly.

**Comment by aaronshurley** [comment]
This will be available in the next version ([v0.41.0](https://github.com/vmware-tanzu/carvel-ytt/releases/tag/v0.41.0)).
